static/src/js/app/components/quick-add-button/QuickAddVocabForm.vue

```diff
@@ -0,0 +1,275 @@
+          (ele) => ele.lang === this.$store.state.me.lang,
+        );
+        this.vocabularyListId = foundLangListID.id || null;
+      }
+    },
+    data() {
+      return {
+        vocabularyListId: null,
+        headword: null,
+        gloss: null,
+        latticeNodeId: null,
+        familiarityRating: 1,
+        showSuccesAlert: false,
+        showUnsuccessfullAlert: false,
+        submitting: false,
+      };
+    },
+    methods: {
+      onRatingChange(rating) {
+        this.familiarityRating = rating;
+      },
+      resetLatticeNodeId() {
+        this.latticeNodeId = null;
+      },
+      async handleSubmit() {
+        this.showUnsuccessfullAlert = false;
+        this.showSuccesAlert = false;
+        this.submitting = true;
+
+        const { headword, gloss, vocabularyListId } = this;
+        if (!headword || !gloss || !vocabularyListId) {
+          this.submitting = false;
+          return;
+        }
+
+        await this.$store.dispatch(CREATE_PERSONAL_VOCAB_ENTRY, {
+          headword,
+          gloss,
+          vocabularyListId,
+          familiarity: this.familiarityRating,
+          node: this.latticeNodeId,
+        });
+        if (this.$store.state.personalVocabAdded) {
+          this.headword = null;
+          this.gloss = null;
+          this.showSuccesAlert = true;
+          this.$store.dispatch(RESET_LATTICE_NODES_BY_HEADWORD);
+        } else {
+          this.showUnsuccessfullAlert = true;
+        }
+        // updates pref language on form submit
+        if (this.vocabularyListId) {
+          const prefLang = this.$store.state.personalVocabLangList.find(
+            (ele) => ele.id === this.vocabularyListId,
+          );
+          if (prefLang.lang !== this.$store.state.me.lang) {
+            await this.$store.dispatch(SET_LANGUAGE_PREF, {
+              lang: prefLang.lang,
+            });
+          }
+        }
+        this.submitting = false;
+      },
+      formatLang(lang) {
+        return LANGUAGES[lang];
+      },
+      setFocus() {
+        this.$nextTick(() => {
+          this.$refs.select.focus();
+        });
+      },
+      // method function for parent components to reset the form when modal is closed
+      resetForm() {
+        this.headword = null;
+        this.gloss = null;
+        this.showSuccesAlert = false;
+        this.showUnsuccessfullAlert = false;
+        this.$store.dispatch(RESET_LATTICE_NODES_BY_HEADWORD);
+      },
+      async getHeadword() {
+        await this.$store.dispatch(FETCH_LATTICE_NODES_BY_HEADWORD, {
+          headword: this.headword,
+        });
+        // sets first latticenode as the default in select options
+        if (this.latticeNode) {
+          this.latticeNodeId = this.latticeNode[0].pk;
+        }
+      },
+      onSelect(event) {
+        // note: '+' changes the string to a number
+        const node = this.latticeNode.find((ele) => +ele.pk === +event.target.value);
```

Comment:
👍  for the note about the `+`. That's a neat trick, although you could make the case that `parseInt()` is clearer and reflects the expected format of the number.

Reply:
ahh that would be more descriptive! will update with the `parseInt()`